parameter_tools/src/main/java/coppercore/parameter_tools/JSONSync.java

```diff
@@ -0,0 +1,115 @@
+package coppercore.parameter_tools;
+
+import com.google.gson.*;
+import java.io.FileReader;
+import java.io.FileNotFoundException;
+
+public class JSONSync<T> {
+
+    private final Gson gson;
+
+    private T instance;
+
+    private String file;
+
+    private final JSONSyncConfig config;
+
+    public T getObject(){
+        return instance;
+    }
+
+    private FileReader getFileReader(String path){
+        try {
+            return new FileReader(path);
+        }catch (FileNotFoundException e){
+            throw new RuntimeException("File not found "+path, e);
+        }
+    }
+    
+    @SuppressWarnings("unchecked")
+    public void loadData(){
+        instance = gson.fromJson(getFileReader(file), (Class<T>) instance.getClass());
+    }
+
+    public void setFile(String newFilePath){
+        file = newFilePath;
+    }
+
+    private Gson generateGson(){
+        return new GsonBuilder().serializeNulls().create();
+    }
+
+    public JSONSync(T instance, String file, JSONSyncConfig config){
+        this.instance = instance;
+        this.gson = generateGson();
+        this.config = (config == null)? new JSONSyncConfigBuilder().build() : config;
+        this.file = file;
+    }
+
+    private static record JSONSyncConfig(boolean serializeNulls, boolean prettyPrinting, boolean excludeFieldsWithoutExposeAnnotation, FieldNamingPolicy namingPolicy,
+    ToNumberPolicy numberToNumberPolicy, ToNumberPolicy objectToNumberPolicy, LongSerializationPolicy longSerializationPolicy, boolean autoReload) {       
+        public JSONSyncConfig(JSONSyncConfigBuilder builder){
+            this(builder.serializeNulls, builder.prettyPrinting, builder.excludeFieldsWithoutExposeAnnotation, builder.namingPolicy, builder.numberToNumberPolicy,
+                builder.objectToNumberPolicy, builder.longSerializationPolicy, builder.autoReload);
+        }
+    }
```

Comment:
Why is there a method within a method here?

Reply:
One is a record which is a data structure that is immutable and the other is a constructor for the record.